kyuubi-server/src/main/scala/org/apache/kyuubi/operation/LaunchEngine.scala

```diff
@@ -37,21 +42,28 @@ class LaunchEngine(session: KyuubiSessionImpl, override val shouldRunAsync: Bool
     setState(OperationState.PENDING)
   }
 
-  override protected def afterRun(): Unit = {
-    OperationLog.removeCurrentOperationLog()
-  }
+  override protected def afterRun(): Unit = { }
 
   override protected def runInternal(): Unit = {
     val asyncOperation: Runnable = () => {
       setState(OperationState.RUNNING)
       try {
-        session.openEngineSession()
+        session.openEngineSession(getOperationLog)
         setState(OperationState.FINISHED)
       } catch {
         onError()
       } finally {
-        // TODO: delay to close it for async mode to enable client to get more launch engine log
-        session.closeOperation(getHandle)
+        new Thread(s"close-launch-engine-op-${getHandle}") {
+          override def run(): Unit = {
+            if (shouldRunAsync) {
+              // delay to close it for async mode to enable client to get more launch engine log
+              Thread.sleep(DEFAULT_LAUNCH_ENGINE_OPERATION_CLOSE_DELAY)
+            }
+            if (!isClosedOrCanceled) {
+              session.closeOperation(getHandle)
+            }
+          }
+        }.start()
```

Comment:
how about we delay deleting the launch log when the related session being closed?

Reply:
It gives more opportunity to for admins to check what happens

Reply:
make senses